data/keystore/src/main/java/tech/pegasys/artemis/bls/keystore/KeyStoreLoader.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.bls.keystore;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import java.io.IOException;
+import java.nio.file.Path;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.artemis.bls.keystore.model.KeyStoreData;
+
+/** Provide utility methods to load/store BLS KeyStore from json format */
+public class KeyStoreLoader {
+  private static final ObjectMapper OBJECT_MAPPER =
+      new ObjectMapper().registerModule(new KeyStoreBytesModule());
+
+  public static KeyStore loadFromJson(final String json) throws Exception {
+    requireNonNull(json);
+    final KeyStoreData keyStoreData = OBJECT_MAPPER.readValue(json, KeyStoreData.class);
+    checkArgument(keyStoreData.getVersion() == 4, "Key Store version 4 is supported.");
+    return new KeyStore(keyStoreData);
+  }
+
+  public static KeyStore loadFromFile(final Path keystoreFile) throws Exception {
+    requireNonNull(keystoreFile);
+    final KeyStoreData keyStoreData =
+        OBJECT_MAPPER.readValue(keystoreFile.toFile(), KeyStoreData.class);
+    checkArgument(keyStoreData.getVersion() == 4, "Key Store version 4 is supported.");
```

Comment:
I'm not sure that an `IllegalArgumentException` is the right thing to do when the format is unsupported. It probably should be the same exception as we wind up throwing for invalid content (with clear message) so it's clear the user input was invalid vs a programmer error passing in values that should never have been passed to this function.

Basically the CLI is going to need to identify these failures as something it should report back to the user in a nice way and it can't do that if it gets very generic exception types.

Applies to the other version checks in this file too.

Reply:
I suspect this is the magic 4 from below that should be a constant somewhere.

Reply:
Updated to use checks and raise custom exception. Also taken care of version constant.